src/routes/courierPage.tsx

```diff
@@ -1,314 +1,314 @@
-import * as React from "react";
-import { createFileRoute, Link } from "@tanstack/react-router";
-import { useState, useEffect } from "react";
-import {
-  getFirestore,
-  collection,
-  addDoc,
-  doc,
-  getDocs,
-  query,
-  where,
-  updateDoc,
-  Timestamp,
-} from "firebase/firestore";
-import Select, { SingleValue } from "react-select";
-import Swal from "sweetalert2";
-
-export const Route = createFileRoute("/courierPage")({
-  component: RouteComponent,
-});
-
-type UserOption = {
-  value: string;
-  label: string;
-};
-
-type CompartmentOption = {
-  id: string;
-  compartment: string | number; // Updated to handle both string and number
-};
-
-function RouteComponent() {
-  const [users, setUsers] = useState<UserOption[]>([]);
-  const [selectedUser, setSelectedUser] =
-    useState<SingleValue<UserOption>>(null);
-  const [compartmentSizes] = useState([
-    { value: "Extra Large", label: "Extra Large" },
-    { value: "Large", label: "Large" },
-    { value: "Medium", label: "Medium" },
-    { value: "Small", label: "Small" },
-  ]);
-  const [selectedSize, setSelectedSize] =
-    useState<SingleValue<{ value: string; label: string }>>(null);
-  const [availableCompartments, setAvailableCompartments] = useState<
-    CompartmentOption[]
-  >([]);
-  const [selectedCompartment, setSelectedCompartment] = useState<string>("");
-  const [selectedCourier, setSelectedCourier] = useState<string>("Amazon");
-
-  const db = getFirestore();
-
-  // Fetch users from Firestore
-  useEffect(() => {
-    const fetchUsers = async () => {
-      const usersCollection = collection(db, "users");
-      const userDocs = await getDocs(usersCollection);
-      const userList = userDocs.docs.map((doc) => ({
-        value: doc.id,
-        label: doc.data().username,
-      }));
-      setUsers(userList);
-    };
-
-    fetchUsers();
-  }, [db]);
-
-  // Fetch compartments when a size is selected
-  useEffect(() => {
-    if (!selectedSize) return;
-
-    const fetchCompartments = async () => {
-      try {
-        const vaultsCollection = collection(db, "vault");
-        const compartmentsQuery = query(
-          vaultsCollection,
-          where("packsize", "==", selectedSize.value),
-          where("status", "==", "available")
-        );
-        const compartmentDocs = await getDocs(compartmentsQuery);
-
-        const compartments = compartmentDocs.docs.map((doc) => ({
-          id: doc.id,
-          compartment: doc.data().compartment,
-        }));
```

Comment:
One suggestion to organizer your code is to write this firebase/firestore related queries into another file. This is helpful in case you need to reuse the logic throughout the application. You can also you a custom hook to handle this interface with firebase.